### PR TITLE
Force Lambda failure

### LINKED ISSUE #555: Force lambda failure
Hi, thanks for this great repo which makes it super easy to deploy web applications to AWS lambdas.
I am using this adapter on top of uvicorn and FastAPI. My problem is that it is almost impossible to make my lambda fail on some specific cases that I want my lambda to fail, instead of returning an http response with status code 500 (or whatever another status code).
Uvicorn will just catch any exception that comes from the application, and will return a response 500 with internal server error message, and there is no way to disable that behaviour.
The only way I could force the lambda to fail is by doing os._exit(1) in my code, which is not the best solution because the app won't get terminated correctly. In addition, once this workaround is done, the next lambda invocation will take around 30 secs to get processed (normally it takes ~5 seconds to run with cold start).
My question and suggestion is to add new environment variable which has the status code in which we want the lambda to fail, and if the adapter gets that status code in the response, it will make the lambda fail instead of returning the response as is.

Thanks!

### PR DESCRIPTION
**Pull request by bnusunny**
*Issue #, if available:*

close #555 #537 

*Description of changes:*

Add `AWS_LWA_ERROR_STATUS_CODES` environment variable, to allow developers to force Lambda failure.

A comma-separated list of HTTP status codes that will cause Lambda invocations to fail. Supports individual codes and ranges (e.g. "500,502-504,422"). When the web application returns any of these status codes, the Lambda invocation will fail and trigger error handling behaviors like retries or DLQ processing. This is useful for treating certain HTTP errors as Lambda execution failures. This feature is disabled by default.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -116,10 +117,33 @@ impl Default for AdapterOptions {
...
+        .flat_map(|part| {
+            let part = part.trim();
+            if part.contains('-') {
+                let range: Vec<&str> = part.split('-').collect();
+                if range.len() == 2 {
+                    if let (Ok(start), Ok(end)) = (range[0].parse::<u16>(), range[1].parse::<u16>()) {
+                        return (start..=end).collect::<Vec<_>>();
+                    }
+                }
+                vec![]
```
Comment (DiscreteTom) [inline src/lib.rs:139]: Maybe better to print a warning if the env var format is incorrect.

File: src/lib.rs [hunk 2]
```diff
@@ -116,10 +117,33 @@ impl Default for AdapterOptions {
...
+            } else {
+                part.parse::<u16>().map_or(vec![], |code| vec![code])
```
Comment (DiscreteTom) [inline src/lib.rs:141]: Same as the above, suggest to add a warning.

File: src/lib.rs (hunk 1 above)
Comment (bnusunny) [reply src/lib.rs:140]: Done

File: src/lib.rs (hunk 2 above)
Comment (bnusunny) [reply src/lib.rs]: Done

**Review by DiscreteTom — APPROVED** [review]

**Pull request merged**